src/platform/graphics/bind-group-format.js

```diff
@@ -87,26 +179,48 @@ class BindStorageTextureFormat extends BindBaseFormat {
 }
 
 /**
- * @ignore
+ * BindGroupFormat is a data structure that defines the layout of resources (buffers, textures,
+ * samplers) used by rendering or compute shaders. It describes the binding points for each
+ * resource type, and the visibility of these resources in the shader stages.
+ * Currently this class is only used on WebGPU platform to specify the input and output resources
+ * for vertex, fragment and compute shaders written in {@link SHADERLANGUAGE_WGSL} language.
+ *
+ * @category Graphics
  */
 class BindGroupFormat {
-    /** @type {BindUniformBufferFormat[]} */
+    /**
+     * @type {BindUniformBufferFormat[]}
+     * @private
+     */
     uniformBufferFormats = [];
 
-    /** @type {BindTextureFormat[]} */
+    /**
+     * @type {BindTextureFormat[]}
+     * @private
+     */
     textureFormats = [];
 
-    /** @type {BindStorageTextureFormat[]} */
+    /**
+     * @type {BindStorageTextureFormat[]}
+     * @private
+     */
     storageTextureFormats = [];
 
-    /** @type {BindStorageBufferFormat[]} */
+    /**
+     * @type {BindStorageBufferFormat[]}
+     * @private
+     */
     storageBufferFormats = [];
 
     /**
+     * Create a new instance.
+     *
      * @param {import('./graphics-device.js').GraphicsDevice} graphicsDevice - The graphics device
      * used to manage this vertex format.
-     * @param {(BindBaseFormat|BindTextureFormat|BindStorageTextureFormat|BindStorageBufferFormat)[]} formats
-     * - An array of bind formats.
+     * @param {(BindTextureFormat|BindStorageTextureFormat|BindUniformBufferFormat|BindStorageBufferFormat)[]} formats
```

Comment:
Could this just be `BindBaseFormat[]`?

Reply:
yeah but I want to avoid people thinking they can specify an instance of the BindBaseFormat too.